Source/st_lib.c

```diff
@@ -221,7 +221,7 @@ void STlib_updatePercent
   char *outrng,             //jff 2/16/98 add color translation to digit output
   int refresh )
 {
-  if (refresh || *per->n.on) // killough 2/21/98: fix percents not updated;
+  if (*per->n.on && refresh) // killough 2/21/98: fix percents not updated;
```

Comment:
Sorry, but this is how it was in Boom 2.01, i.e. before it got "fixed":
https://github.com/Doom-Utils/historic-ports/blame/ce8bbd699f29452edf7d02b912b49146a80144ed/st_lib.c#L184

Looking at `STlib_updateNum()` and related functions, it seems that the `refresh` value is outright ignored there. So, I am all for rolling back to your initial approach of removing the check for `refresh` - but please add a comment.

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/216/commits/6446fe0266ada9f3c536925118f6c5d63273943a